bluepysnap/simulation.py

```diff
@@ -125,8 +136,17 @@ def simulator(self):
     @cached_property
     def node_sets(self):
         """Returns the NodeSets object bound to the simulation."""
-        path = self.to_libsonata.node_sets_file
-        return NodeSets.from_file(path) if path else NodeSets.from_dict({})
+        try:
+            path = self.circuit.to_libsonata.node_sets_path
+        except BluepySnapError:
```

Comment:
I see that `self.circuit` might raise BluepySnapError if the circuit config file doesn't exist.
Is the simulation config considered valid even in this case?
If we need to keep the check, a comment with the type of expected error may be useful.

Reply:
Not commenting on the validity of the simulation but, according to the spec, `network` is an optional entry, so I thought it'd be better to not require it being there for the node sets, either.

-> Added a comment